src/ztc/api/utils/validators.py

```diff
@@ -28,9 +28,17 @@ def __init__(self, relation_field: str, catalogus_field="catalogus"):
         self.relation_field = relation_field
         self.catalogus_field = catalogus_field
 
+    def set_context(self, serializer):
+        """
+        This hook is called by the serializer instance,
+        prior to the validation call being made.
+        """
+        # Determine the existing instance, if this is an update operation.
+        self.instance = getattr(serializer, "instance", None)
+
     def __call__(self, attrs: dict):
         relations = attrs.get(self.relation_field)
-        catalogus = attrs.get(self.catalogus_field)
+        catalogus = attrs.get(self.catalogus_field) or self.instance.catalogus
```

Comment:
If it doesn't pass field validation, this validator won't even be triggered right?

Reply:
yeah, you're right, it may only run if it's specified as a field validator, DRF tries to run as much validation in one go to bundle errors together instead of bailing out at the first validation error. But serializer-level validation only runs if all field validations passed